programs/clearing_house/src/controller/pnl.rs

```diff
@@ -254,49 +254,57 @@ pub fn settle_expired_position(
         "User must first burn lp shares for expired market"
     )?;
 
-    let (base_asset_value, unrealized_pnl) = calculate_base_asset_value_and_pnl_with_expiry_price(
+    let base_asset_value = calculate_base_asset_value_with_expiry_price(
         &user.perp_positions[position_index],
         perp_market.expiry_price,
     )?;
 
+    let base_asset_amount = user.perp_positions[position_index].base_asset_amount;
+    let quote_entry_amount = user.perp_positions[position_index].quote_entry_amount;
+
+    let position_delta = PositionDelta {
+        quote_asset_amount: base_asset_value,
+        base_asset_amount: -user.perp_positions[position_index].base_asset_amount,
+    };
+
+    update_position_and_market(
+        &mut user.perp_positions[position_index],
+        perp_market,
+        &position_delta,
+    )?;
+
     let fee = base_asset_value
-        .checked_mul(fee_structure.fee_tiers[0].fee_numerator as u128)
+        .checked_mul(fee_structure.fee_tiers[0].fee_numerator as i64)
         .ok_or_else(math_error!())?
-        .checked_div(fee_structure.fee_tiers[0].fee_denominator as u128)
+        .checked_div(fee_structure.fee_tiers[0].fee_denominator as i64)
         .ok_or_else(math_error!())?;
 
-    let unrealized_pnl_with_fee = unrealized_pnl
-        .checked_sub(cast_to_i128(fee)?)
-        .ok_or_else(math_error!())?;
-
-    let pnl_to_settle_with_user = update_pnl_pool_and_user_balance(
+    update_quote_asset_amount(
+        &mut user.perp_positions[position_index],
         perp_market,
-        quote_spot_market,
-        user,
-        unrealized_pnl_with_fee,
+        -fee.abs(),
     )?;
 
-    update_user_settled_pnl(user, unrealized_pnl_with_fee.cast()?)?;
+    let pnl = user.perp_positions[position_index].quote_asset_amount;
```

Comment:
should we change this to `get_claimable_pnl()`?

Reply:
not sure what you mean